File: 12-hash/1-intro/main.cpp
// Hashing - C++ Example

#include <iostream>
#include <unordered_map>

int main() {
    std::unordered_map<std::string, int> hash_map;

    // Insert
    hash_map["apple"] = 3;
    hash_map["banana"] = 5;

    // Lookup
    std::cout << "Apple count: " << hash_map["apple"] << std::endl;

    // Update
    hash_map["apple"] += 2;
    std::cout << "Updated apple count: " << hash_map["apple"] << std::endl;

    return 0;
}

File: 12-hash/1-intro/main.java
// Hashing - Java Example

import java.util.HashMap;

class HashingIntro {

    public static void main(String[] args) {
        HashMap<String, Integer> hashMap = new HashMap<>();

        // Insert
        hashMap.put("apple", 3);
        hashMap.put("banana", 5);

        // Lookup
        System.out.println("Apple count: " + hashMap.getOrDefault("apple", 0));

        // Update
        hashMap.put("apple", hashMap.get("apple") + 2);
        System.out.println("Updated apple count: " + hashMap.get("apple"));
    }
}
